docs/ruleset_functions/get_number_of_floors.md

```diff
@@ -0,0 +1,27 @@
+# get_number_of_floors
+**Schema Version:** 0.0.22  
+
+**Description:** gets the number of floors in the building
+
+**Inputs:**
+- **RMD**
+
+**Returns:**  
+- **result**: an integer indicating the number of floors
+ 
+**Function Call:**
+
+## Logic:
+- create a set of the floor names.  We use a set because only unique elements can be added to the set: `floor_names = set()`
+- loop through the building segments: `for segment in RMD.building.building_segments:`
+	- loop through the zones: `for zone in segment.zones:`
+		- add the floor name to the set of floor names.  If the floor name is already in the set, nothing happens at this step: `floor_names.add(zone.floor_name)`
+- the number of items in the set equals the number of floors in the building.  Set result equal to the length of the set: `result = len(floor_names)`
+
```

Comment:
we need the number of floors in the entire building.  We are expecting multiple building segments on one floor, and simply returning the BuildingSegment.number_of_floors_above_grade doesn't tell us whether we are double-counting

> Why not use BuildingSegment.number_of_floors_above_grade+BuildingSegment.number_of_floors_below_grade?

Reply:
Not sure using the floor name would work. In eQUEST, side-by-side portion of the building may be modeled as two separate shells and each floor within a shell will have a unique name even if it has the same elevation as another floor object. Perhaps add number of floors to Building? Or add Zone.elevation or Space.elevation? Let's discuss on the call.

Reply:
updated the function logic based on new schema variables: `zone.elevation` and `zone.ceiling_height`

Reply:
@mkarpman and @KarenWGard I am not sure what was discussed and the reasoning for going with zone.elevation instead of number of above grade and below grade floors at the building level. It would help to discuss that during the call today

Reply:
The logic above still refers to floor names, so I am confused. I thought we decided to use zone.elevation criteria?

Reply:
Per the call on 5/1/2023 - I have emailed Mike Rosenberg to ask if he has a consistent definition of the way to define number of floors.  If we don't have a clear and consistent definition, revert to using floor names.